packages/client/modules/teamDashboard/components/ProviderRow/MattermostProviderRow.tsx

```diff
@@ -0,0 +1,159 @@
+import React, {useState} from 'react'
+import styled from '@emotion/styled'
+import {useFragment} from 'react-relay'
+import {MattermostProviderRow_viewer$key} from '~/__generated__/MattermostProviderRow_viewer.graphql'
+import graphql from 'babel-plugin-relay/macro'
+import FlatButton from '../../../../components/FlatButton'
+import Icon from '../../../../components/Icon'
+import ProviderCard from '../../../../components/ProviderCard'
+import ProviderActions from '../../../../components/ProviderActions'
+import RowInfo from '../../../../components/Row/RowInfo'
+import RowInfoCopy from '../../../../components/Row/RowInfoCopy'
+import {MenuPosition} from '../../../../hooks/useCoords'
+import useMenu from '../../../../hooks/useMenu'
+import {PALETTE} from '../../../../styles/paletteV3'
+import {ICON_SIZE} from '../../../../styles/typographyV2'
+import {Breakpoint, Layout, Providers} from '../../../../types/constEnums'
+import useMutationProps, {MenuMutationProps} from '../../../../hooks/useMutationProps'
+import useBreakpoint from '../../../../hooks/useBreakpoint'
+import MattermostPanel from './MattermostPanel'
+import MattermostProviderLogo from '../../../../components/MattermostProviderLogo'
+import MattermostSVG from '../../../../components/MattermostSVG'
+import MattermostConfigMenu from './MattermostConfigMenu'
+
+const StyledButton = styled(FlatButton)({
+  borderColor: PALETTE.SLATE_400,
+  color: PALETTE.SLATE_700,
+  fontSize: 14,
+  fontWeight: 600,
+  minWidth: 36,
+  paddingLeft: 0,
+  paddingRight: 0,
+  width: '100%'
+})
+
+interface Props {
+  teamId: string
+  viewerRef: MattermostProviderRow_viewer$key
+}
+
+const MenuButton = styled(FlatButton)({
+  color: PALETTE.GRAPE_700,
+  fontSize: ICON_SIZE.MD18,
+  height: 24,
+  userSelect: 'none',
+  marginLeft: 4,
+  padding: 0,
+  width: 24
+})
+
+const StyledIcon = styled(Icon)({
+  fontSize: ICON_SIZE.MD18
+})
+
+const ListAndMenu = styled('div')({
+  display: 'flex',
+  position: 'absolute',
+  right: 16,
+  top: 16
+})
+
+const MattermostLogin = styled('div')({})
+
+const ProviderName = styled('div')({
+  color: PALETTE.SLATE_700,
+  fontSize: 16,
+  fontWeight: 600,
+  lineHeight: '24px',
+  alignItems: 'center',
+  display: 'flex',
+  marginRight: 16,
+  verticalAlign: 'middle'
+})
+
+const CardTop = styled('div')({
+  display: 'flex',
+  justifyContent: 'flex-start',
+  padding: Layout.ROW_GUTTER
+})
+
+const ExtraProviderCard = styled(ProviderCard)({
+  flexDirection: 'column',
+  padding: 0
+})
+
+const MattermostProviderRow = (props: Props) => {
+  const {viewerRef, teamId} = props
+  const viewer = useFragment(
+    graphql`
+      fragment MattermostProviderRow_viewer on User {
+        ...MattermostProviderRowViewer @relay(mask: false)
```

Comment:
here we've got a User fragment nested in a User fragment. We can simplify by removing the nesting.
 
let's inline the fragment below so we get

```gql
fragment MattermostProviderRow_viewer on User {
        ...MattermostPanel_viewer
        teamMember(teamId: $teamId) {
          integrations {
            mattermost {
              isActive
            }
          }
        }
      }
```

then in the Remove mutation, we just reference `MattermostProviderRow_viewer` instead

Reply:
This bugged me too!

I copied the pattern from here: https://github.com/ParabolInc/parabol/blob/2f90cb3592e9fedd8f0badcce9db90a6150dae5f/packages/client/modules/teamDashboard/components/ProviderRow/SlackProviderRow.tsx#L155

I figured, you had your reasons 😉 

I've gone ahead and made this a single fragment (but left Slack alone, for now)